### PR TITLE
add KNET documentation and known issues

### PR DESCRIPTION
**Pull request by KanjiMonster**
Add some documentation on how to enable KNET interfaces and their known issues.

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>

### CODE REVIEW

File: platform_configuration/bcm_knet_interfaces.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+---
+title: Broadcom KNET network interfaces
+parent: Platform Configuration
+nav_order: 3
```
Comment (jklare) [inline platform_configuration/bcm_knet_interfaces.md:4]:
```suggestion
nav_order: 6
```

File: platform_configuration/bcm_knet_interfaces.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+---
+
+Broadcom KNET network interfaces are an alternative implementation for the representation of the switch ports. They directly forward the packets in the kernel to and from the switch, reducing overhead and improving throughput and latency when communicating from the switch itself.
```
Comment (jklare) [inline platform_configuration/bcm_knet_interfaces.md:7]:
```suggestion
Broadcom KNET network interfaces ("bcm-knet") are an alternative for the "tun" interface type representation of the switch ports. They directly forward the packets in the kernel to and from the switch, reducing overhead and improving throughput and latency when communicating from the switch itself.
```

File: platform_configuration/bcm_knet_interfaces.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+Due to the nature of the implementation, these can only be used when running `baseboxd` on the switch itself. When running `baseboxd` off-switch, only the tap interfaces are available.
+
+**Note**: Broadcom KNET network interfaces is still an experimental feature, and may not always work as expected.
```
Comment (jklare) [inline platform_configuration/bcm_knet_interfaces.md:11]:
```suggestion
**Note**: The use of Broadcom KNET network interfaces is still an experimental feature, and may not always work as expected. 
```

**Review by jklare — CHANGES_REQUESTED** [review]
lgtm, a couple of suggestions

**Comment by KanjiMonster** [comment]
Updated, and added a link to the known issues to encourage users to look there before reporting issues.

**Review by jklare — APPROVED** [review]
lgtm

**Pull request merged**